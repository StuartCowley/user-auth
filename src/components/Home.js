import React from "react";
import { Link } from "react-router-dom";

import "../styles/home.css"

const Home = ({ user }) => {
  return (
    <div className="home">
      {user.firstName ? (
        <h2>Welcome to you, <span className="home__username">{user.firstName}</span></h2>
      ) : (
        <h2>
          You are not logged in, <Link to="login">log in</Link> or <Link to="signup">signup</Link>
        </h2>
      )}
    </div>
  )
}

export default Home;