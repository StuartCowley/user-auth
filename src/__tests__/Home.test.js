import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from '../components/Home';

describe("Home component", () => {
  const validProps = {
    user: {
      firstName: 'test first name'
    }
  };

  it("matches snapshot", () => {
    const generated = renderer
    .create(<Router><Home user={validProps.user} /></Router>)
    .toJSON();

  expect(generated).toMatchSnapshot();
  });
  
  it("renders welcome message if user logged in", () => {
    render(<Router><Home user={validProps.user}/></Router>);
    const message = screen.getByRole("heading", /Welcome to you, test first name/);

    expect(message).toBeInTheDocument;
  });

  it("renders redirect message if user not logged in", () => {
    const validProps = {
      user: {
        firstName: ""
      }
    }
    render(<Router><Home user={validProps.user} /></Router>);
    const message = screen.getByRole("heading", /You are not logged in, log in or signup/);

    expect(message).toBeInTheDocument;
  });
});