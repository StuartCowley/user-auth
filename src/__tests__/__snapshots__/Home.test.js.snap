// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div
  className="home"
>
  <h2>
    Welcome to you, 
    <span
      className="home__username"
    >
      test first name
    </span>
  </h2>
</div>
`;
